github/VisualStudio
# Use modern clone dialog when re-cloning a repository

_Depends on #2198_

Now that the clone dialog can clone any arbitrary GitHub URL (not just ones owned by the user), we don't need a separate dialog when re-cloning a URL using the `Start Page`.

### What this PR does

- Use the same clone dialog when cloning a new repository or re-cloning one that has been cloned before
- Remove the obsolete `ShowReCloneDialog` API and related classes

### How to test

1. Open the `Start Page` (click `File > Start Page`)
2. Click on the `GitHub` button which is under `Open`
3. Clone a repository
4. Delete the repository folder (necessary to make the URL link appear)
5. Open the `Start Page` (click `File > Start Page`)
6. Click on the `https://` link to the repository you cloned then deleted
7. `Open from GitHub` dialog should appear

Fixes #2199
